core/metacling/src/TCling.cxx

```diff
@@ -1258,7 +1274,32 @@ TCling::TCling(const char *name, const char *title)
       // Load libc and stl first.
       LoadModules({"libc", "stl"}, *fInterpreter);
 
-      LoadModules({"ROOT_Foundation_C", "ROOT_Config", "ROOT_Foundation_Stage1_NoRTTI", "Core", "RIO"}, *fInterpreter);
+      // Load core modules
+      std::vector<std::string> CoreModules = {"ROOT_Foundation_C","ROOT_Config","ROOT_Foundation_Stage1_NoRTTI", "Core", "RIO"};
+      LoadModules(CoreModules, *fInterpreter);
+
+      // Otherwise module conflicts between rootcling and root
+      if (!fromRootCling) {
```

Comment:
This check can likely go away if we check if the module is already built (i.e. if it has an on-disk representing pcm). This way rootcling will become more efficient by building a module, while reusing the previously build ones.